[CI] DocsClientYamlTestSuiteIT test {yaml=reference/search/search-your-data/search-your-data/line_118} failing

**Build Scans:**
- [elasticsearch-periodic #4201 / openjdk23_java-matrix](https://gradle-enterprise.elastic.co/s/ox2rdcz4v4kc6)
- [elasticsearch-periodic #4197 / openjdk23_java-matrix](https://gradle-enterprise.elastic.co/s/cjnr4ybtf3vhu)
- [elasticsearch-periodic #4193 / openjdk23_java-matrix](https://gradle-enterprise.elastic.co/s/a4tk64oyfy5ju)
- [elasticsearch-periodic #4189 / openjdk23_java-matrix](https://gradle-enterprise.elastic.co/s/kylg5jatnnpha)
- [elasticsearch-periodic #4185 / openjdk23_java-matrix](https://gradle-enterprise.elastic.co/s/fneau2lg4bogk)
- [elasticsearch-periodic #4181 / openjdk23_java-matrix](https://gradle-enterprise.elastic.co/s/sv7jhxpueldic)
- [elasticsearch-periodic #4177 / openjdk23_java-matrix](https://gradle-enterprise.elastic.co/s/3ksheoiu5jdp2)
- [elasticsearch-periodic #4173 / openjdk23_java-matrix](https://gradle-enterprise.elastic.co/s/5dc6jsemskmp2)
- [elasticsearch-periodic #4169 / openjdk23_java-matrix](https://gradle-enterprise.elastic.co/s/sgpofoctmxmee)
- [elasticsearch-periodic #4165 / openjdk23_java-matrix](https://gradle-enterprise.elastic.co/s/ficwmozeog4nw)

**Reproduction Line:**
```
./gradlew ':docs:yamlRestTest' --tests "org.elasticsearch.smoketest.DocsClientYamlTestSuiteIT.test {yaml=reference/search/search-your-data/search-your-data/line_118}" -Dtests.seed=D7D311EF0CD4361A -Dtests.locale=fr-MC -Dtests.timezone=Antarctica/Casey -Druntime.java=23
```

**Applicable branches:**
7.17

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('test%20%7Byaml%3Dreference%2Fsearch%2Fsearch-your-data%2Fsearch-your-data%2Fline_118%7D'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.smoketest.DocsClientYamlTestSuiteIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: Failure at [reference/search/search-your-data/search-your-data:191]: Expected a map containing
                _shards: a map containing
                    total: <1>
               successful: <1>
                  skipped: <0>
                   failed: <0>
                   hits: a map containing
                    total: a map containing
                      value: <5>
                   relation: "eq"
                max_score: <1.0>
                     hits: a list containing
                          0: a map containing
                       _index: "my-index-000001"
                        _type: "_doc"
                          _id: "0"
                       _score: <1.0>
                      _source: a map containing
                     @timestamp: "2099-11-15T14:12:12"
                           http: a map containing
                          request: a map containing
                             method: "get"
                         response: a map containing
                        status_code: <200>
                              bytes: <1070000>
                          version: "1.1"
                         source: a map containing
                               ip: "127.0.0.1"
                        message: "GET /search HTTP/1.1 200 1070000"
                           user: a map containing
                               id: "kimchy"
                          1: a map containing
                       _index: "my-index-000001"
                        _type: "_doc"
                          _id: "1"
                       _score: <1.0>
                      _source: a map containing
                     @timestamp: "2099-11-15T14:12:12"
                           http: a map containing
                          request: a map containing
                             method: "get"
                         response: a map containing
                        status_code: <200>
                              bytes: <1070000>
                          version: "1.1"
                         source: a map containing
                               ip: "10.42.42.42"
                        message: "GET /search HTTP/1.1 200 1070000"
                           user: a map containing
                               id: "elkbee"
                          2: a map containing
                       _index: "my-index-000001"
                        _type: "_doc"
                          _id: "2"
                       _score: <1.0>
                      _source: a map containing
                     @timestamp: "2099-11-15T14:12:12"
                           http: a map containing
                          request: a map containing
                             method: "get"
                         response: a map containing
                        status_code: <200>
                              bytes: <1070000>
                          version: "1.1"
                         source: a map containing
                               ip: "10.42.42.42"
                        message: "GET /search HTTP/1.1 200 1070000"
                           user: a map containing
                               id: "elkbee"
                          3: a map containing
                       _index: "my-index-000001"
                        _type: "_doc"
                          _id: "3"
                       _score: <1.0>
                      _source: a map containing
                     @timestamp: "2099-11-15T14:12:12"
                           http: a map containing
                          request: a map containing
                             method: "get"
                         response: a map containing
                        status_code: <200>
                              bytes: <1070000>
                          version: "1.1"
                         source: a map containing
                               ip: "10.42.42.42"
                        message: "GET /search HTTP/1.1 200 1070000"
                           user: a map containing
                               id: "elkbee"
                          4: a map containing
                       _index: "my-index-000001"
                        _type: "_doc"
                          _id: "4"
                       _score: <1.0>
                      _source: a map containing
                     @timestamp: "2099-11-15T14:12:12"
                           http: a map containing
                          request: a map containing
                             method: "get"
                         response: a map containing
                        status_code: <200>
                              bytes: <1070000>
                          version: "1.1"
                         source: a map containing
                               ip: "10.42.42.42"
                        message: "GET /search HTTP/1.1 200 1070000"
                           user: a map containing
                               id: "elkbee"
                   took: <4>
              timed_out: <false>
           aggregations: a map containing
              day_of_week: a map containing
doc_count_error_upper_bound: <0>
        sum_other_doc_count: <0>
                    buckets: a list containing
                            0: a map containing
                      doc_count: <5>
                            key: expected "Sunday" but was "Sun"
```

**Issue Reasons:**
- [7.17] 22 consecutive failures in step openjdk23_java-matrix
- [7.17] 22 failures in test test {yaml=reference/search/search-your-data/search-your-data/line_118} (2.6% fail rate in 861 executions)
- [7.17] 22 failures in step openjdk23_java-matrix (100.0% fail rate in 22 executions)
- [7.17] 22 failures in pipeline elasticsearch-periodic (100.0% fail rate in 22 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/docs/reference/search/search-your-data/search-your-data.asciidoc
+++ b/docs/reference/search/search-your-data/search-your-data.asciidoc
@@ -124,7 +124,7 @@ GET /my-index-000001/_search
       "script": {
         "source":
         """emit(doc['@timestamp'].value.dayOfWeekEnum
-        .getDisplayName(TextStyle.FULL, Locale.ROOT))"""
+        .getDisplayName(TextStyle.FULL, Locale.ENGLISH))"""
       }
     }
   },

--- a/x-pack/plugin/eql/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/eql/10_basic.yml
+++ b/x-pack/plugin/eql/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/eql/10_basic.yml
@@ -13,7 +13,7 @@ setup:
                 day_of_week:
                   type: keyword
                   script:
-                    source: "emit(doc['@timestamp'].value.dayOfWeekEnum.getDisplayName(TextStyle.FULL, Locale.ROOT))"
+                    source: "emit(doc['@timestamp'].value.dayOfWeekEnum.getDisplayName(TextStyle.FULL, Locale.ENGLISH))"
   - do:
       bulk:
         refresh: true

--- a/x-pack/plugin/eql/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/eql/20_runtime_mappings.yml
+++ b/x-pack/plugin/eql/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/eql/20_runtime_mappings.yml
@@ -9,7 +9,7 @@ setup:
                 day_of_week:
                   type: keyword
                   script:
-                    source: "emit(doc['@timestamp'].value.dayOfWeekEnum.getDisplayName(TextStyle.FULL, Locale.ROOT))"
+                    source: "emit(doc['@timestamp'].value.dayOfWeekEnum.getDisplayName(TextStyle.FULL, Locale.ENGLISH))"
   - do:
       bulk:
         refresh: true

